pkg/controllers/multicluster/remoteendpointslice_controller.go

```diff
@@ -0,0 +1,237 @@
+		}
+	}
+
+	return nil
+}
+
+func (r *RemoteEndpointSliceReconciler) syncRemoteEndpointSlices(ctx context.Context,
+	service, namespace string, log logr.Logger) error {
+
+	epsSel := labels.NewSelector()
+	r1, _ := labels.NewRequirement(discoveryv1beta1.LabelServiceName, selection.Equals, []string{service})
+	// avoid circular reference, exclude the EndpointSlices created by hybridnet
+	r2, _ := labels.NewRequirement(discoveryv1beta1.LabelManagedBy, selection.NotEquals, []string{remoteEndpointSliceControllerName})
+	epsSel = epsSel.Add(*r1, *r2)
+
+	targetEpsList := &discoveryv1beta1.EndpointSliceList{}
+	if err := r.List(ctx, targetEpsList, client.MatchingLabelsSelector{
+		Selector: epsSel,
+	}, client.InNamespace(namespace)); err != nil {
+		return fmt.Errorf("unable to get endpoint slices of remote cluster %v for service %v/%v: %v",
+			r.ClusterName, service, namespace, err)
+	}
+
+	existRepsList := &multiclusterv1.RemoteEndpointSliceList{}
+	if err := r.ParentCluster.GetClient().List(ctx, existRepsList, client.MatchingLabels(generateRemoteEndpointSliceLabels(
+		service, r.ClusterName))); err != nil {
+		return fmt.Errorf("unable to get remote endpoint slices of remote cluster %v for service %v/%v: %v",
+			r.ClusterName, service, namespace, err)
+	}
+
+	// TODO: Do we need a name prefix here for RemoteEndpointSlices?
+	targetEpsMap := map[string]string{}
+	for _, eps := range targetEpsList.Items {
+		targetEpsMap[eps.Name] = eps.Name
+	}
+
+	for _, reps := range existRepsList.Items {
+		if _, exist := targetEpsMap[reps.Name]; !exist {
+			if err := client.IgnoreNotFound(r.ParentCluster.GetClient().Delete(ctx, &reps)); err != nil {
+				return fmt.Errorf("unable to delete remote endpoint slice %v/%v: %v", reps.Name, reps.Namespace, err)
+			}
+		}
+	}
+
+	for index := range targetEpsList.Items {
+		eps := targetEpsList.Items[index]
+
+		var remoteEndpointSlice = &multiclusterv1.RemoteEndpointSlice{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: eps.Name,
+			},
+		}
+
+		operationResult, err := controllerutil.CreateOrPatch(ctx, r.ParentCluster.GetClient(), remoteEndpointSlice, func() error {
+			if !remoteEndpointSlice.DeletionTimestamp.IsZero() {
+				return fmt.Errorf("endpoint slice %s is terminating, can not be updated", remoteEndpointSlice.Name)
+			}
+
+			remoteEndpointSlice.Labels = eps.Labels
+			remoteEndpointSlice.Annotations = eps.Annotations
+			remoteEndpointSlice.Labels[discoveryv1beta1.LabelManagedBy] = remoteEndpointSliceControllerName
+			remoteEndpointSlice.Labels[constants.LabelRemoteCluster] = r.ClusterName
+
+			remoteEndpointSlice.Spec.RemoteService.Cluster = r.ClusterName
+			remoteEndpointSlice.Spec.RemoteService.Namespace = eps.Namespace
+			remoteEndpointSlice.Spec.RemoteService.Name = service
+
+			remoteEndpointSlice.Spec.AddressType = eps.AddressType
+			remoteEndpointSlice.Spec.Endpoints = eps.Endpoints
+			remoteEndpointSlice.Spec.Ports = eps.Ports
+
```

Comment:
do we need to set ownerReference to parent cluster here?

Reply:
done